contracts/gov/src/migration.rs

```diff
@@ -41,5 +79,42 @@ pub fn migrate_config(
         proposal_deposit: legacy_config.proposal_deposit,
         snapshot_period: legacy_config.snapshot_period,
         anchor_voting_escrow,
-    })
+        voter_weight,
+    })?;
+
+    let legacy_state: LegacyState = read_legacy_state(storage)?;
+
+    state_store(storage).save(&State {
+        contract_addr: legacy_state.contract_addr,
+        poll_count: legacy_state.poll_count,
+        total_share: legacy_state.total_share,
+        total_deposit: legacy_state.total_deposit,
+        pending_voting_rewards: Uint128::zero(),
+    })?;
+
+    for poll_id in 1..=legacy_state.poll_count {
```

Comment:
@cdocu11 migration looks good to me! @iLoveBusinessDevelopment mind confirming this migration logic?

small note - let's rename `migrate_config` to something that makes sense since we are also migrating `poll` here. and/or we can have a separate `migrate_poll` function.